plugins/org.jkiss.dbeaver.data.transfer/src/org/jkiss/dbeaver/tools/transfer/stream/exporter/DataExporterHTML.java

```diff
@@ -167,7 +167,6 @@ private void writeTableTitle(String value, int columns) {
         if (value == null) {
             out.write("&nbsp;");
         } else {
-            value = value.replace("<", "&lt;").replace(">", "&gt;").replace("&", "&amp;");
```

Comment:
Non-escaped `<` and `>` may break HTML and allow potential injection.

The problem here is that these special characters are escaped first and the produced `&` characters are replaced by `replace("&", "&amp;")` too, resulting in broken escape sequence: `<&` &rArr; `&lt;&` &rArr; `&amp;lt;&amp;`.

So the fix is to move `replace("&", "&amp;")` to be the beginning, just before other ones.

Reply:
Thank you for the detailed explanation.